src/wallet/interfaces.cpp

```diff
@@ -123,52 +123,54 @@ WalletTxOut MakeWalletTxOut(const CWallet& wallet,
 namespace CoinJoin = interfaces::CoinJoin;
 class CoinJoinImpl : public CoinJoin::Client
 {
-    std::shared_ptr<CCoinJoinClientManager> m_manager;
+    CCoinJoinClientManager& m_manager;
```

Comment:
why'd you make this a ref?

Reply:
No reason to leave it as a pointer, if we can de-reference the pointer as early as possible, I don't see why not. Keeping the pointer container as a member either indicates that

* The object is responsible for construction/destruction
* The non-dereferenced state needs to be passed as object initialization precedes pointer initialization but dereferencing of that member occurs succeeds initialization, making this acceptable from a deglobalization standpoint but less than ideal from a strict order of initialization standpoint.

  An example would be the passage of `CChainLocksHandler`, `CInstantSendManager` and `CQuorumBlockProcessor`'s respective `unique_ptr` containers to `CChainState` ([source](https://github.com/dashpay/dash/blob/13c12406ae076b58cba580aa47c3419ef5bec361/src/validation.h#L595-L597)).

`CoinJoinImpl` meets neither criteria and so there's little reason to use the pointer container as a member when a reference will do.